docs/Versioning.md

```diff
@@ -53,13 +66,33 @@ As a general principle, implementations SHOULD be robust in the face of missing
+Because onnx.proto is expected to be consumed by multiple independent developers, changes to onnx.oroto SHOULD NOT break code that depends on generated language bindings (e.g., changing the type of an existing field).
 
 ISSUE: define type compatiblity rules either here or under model versioning - probably here
 
 ## Operator versioning
 
-ISSUE: sort out the design and then document.
+ONNX is defined such that the IR can evolve independently from the set of operators. In ONNX, operators represent both the signature and semantics of a given operation.  Operators are abstract interfaces in that they do not imply a specific implementation; rather, they are simply the contract between a model author and the implementations that model may execute on. 
+
+A given operator is identified by a three-tuple: (domain, op_type, op_version), written domain.op_type:op_version in prose (e.g., com.acme.FastConv:3).  Nodes in graphs always refer
+to operators by their three-part identifier.
+
+Once an operator is published, all implementations of that operator's (domain, op_type, op_version) MUST adhere to the signature and semantics of the operator at the time of publication. 
+Any change of semantics implies a new operator, which MAY share a domain and op_type with another operator. This includes adding new behavior triggered only by previously unspecified inputs or attributes - these changes in semantics also MUST have a distinct operator id.
+
+ONNX uses operator sets to group together immutable operator specifications. An ONNX operator set
+specifies both the domain of all operators it includes, as well as an opset version. The opset version is largely independent from the version field of the operators it includes. When the enventory of a given operator set changes either by addition or removal, its opset version MUST increase. Moreover,
+the opset version MUST be no less than the highest operator version number in the set.
+
+ONNX models declare which operator sets they require as a list of two part operator ids (domain, opset_version).  The empty string ("") domain indicates the operators defined as part of the 
+ONNX specification. The union of the operator sets specified by a given model MUST have have have a compatible operator declaration for each node in the model's graph.  
+
+
+How nodes bind to operator declarations is strictly defined and are designed to increase model compatibilitey across ONNX implementations (appealing to the conservative clause of the robustnes principle). 
```

Comment:
s/compatibilitey/compatibility/

Reply:
Thank goodness you are part of the mechanical turk of typo corrections.

Reply:
s/oroto/proto/
s/enventory/inventory/
s/have have have/have/
s/seriaized/serialized/

Reply:
s/therefor/therefore/